src/Selenium2Library/keywords/screenshot.py

```diff
@@ -130,11 +139,24 @@ def _get_screenshot_directory(self):
     def _restore_screenshot_directory(self):
         self.screenshot_root_directory = self._screenshot_path_stack.pop()
 
-    def _get_screenshot_paths(self, filename):
-        filename = filename.format(
-            index=self._get_screenshot_index(filename))
-        filename = filename.replace('/', os.sep)
+    def _get_screenshot_paths(self, filename_template):
         screenshotdir = self._get_screenshot_directory()
+
+        filename = filename_template.format(
+            index=self._get_screenshot_index(filename_template))
+
+        # Sorry for the gnarley regular expression.  it attempts to
+        # match python formatter syntax such as {index} or {index:...}
+        # but not {{index}} or # {{index:...}}
+        if re.search(r'(?<!{){index(![rs])?(:.*?)?}(?!})', filename_template):
+            # make sure the computed filename doesn't exist. We only
+            # do this if the template had the {index} formatting
+            # sequence (or one of it's variations)
```

Comment:
Comments are OK, but perhaps not this much, perhaps if we comment what the regex is trying to match? Regex itself is OK and code is readable.